### ISSUE TITLE
Occasional crash related to push

### ISSUE THREAD

**Issue by jayb-g** [thunderbird/thunderbird-android#8242]
Maybe at the time of network change.

```
type: crash
osVersion: google/shiba/shiba:14/AP2A.240905.003/2024092900:user/release-keys
package: com.fsck.k9:39005
process: com.fsck.k9
processUptime: 53541 + 412 ms
installer: dev.imranr.obtainium

java.util.ConcurrentModificationException
	at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:1061)
	at java.util.LinkedHashMap$LinkedKeyIterator.next(LinkedHashMap.java:1084)
	at com.fsck.k9.controller.push.PushController.updatePushEnabledListeners(PushController.kt:309)
	at com.fsck.k9.controller.push.PushController.updatePushers(PushController.kt:214)
	at com.fsck.k9.controller.push.PushController.access$updatePushers(PushController.kt:34)
	at com.fsck.k9.controller.push.PushController$launchUpdatePushers$1.invokeSuspend(PushController.kt:153)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@6d2bb45, java.util.concurrent.Executors$FinalizableDelegatedExecutorService@926f29a]
```

**Comment by cketti** [comment]
This will be fixed by #8225 and should be included in 8.0b2.

**Comment by cketti** [comment]
Apparently I missed your description and didn't think of other ways than removing an account to trigger that crash. Changing the network is just another way to trigger the same crash.
